simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/actions/Plug.java

```diff
@@ -30,34 +31,34 @@
     @Override
     public void perform(StationState stationState, StationMessageSender stationMessageSender) {
 
-        if (stationState.getConnectorState(connectorId) != ConnectorStatus.UNPLUGGED) {
+        if (stationState.getCableStatus(connectorId) != CableStatus.UNPLUGGED) {
             throw new IllegalStateException(String.format("Connector is not available: %s", connectorId));
         }
 
-        Integer evseId = stationState.findEvseId(connectorId);
+        Evse evse = stationState.findEvseByConnectorId(connectorId);
 
-        if (!stationState.hasOngoingTransaction(evseId)) {
+        if (!evse.hasOngoingTransaction()) {
             Integer transactionId = TransactionIdGenerator.getInstance().getAndIncrement();
-            stationState.findEvse(evseId).createTransaction(transactionId);
+            evse.createTransaction(transactionId);
         }
 
-        stationState.plug(connectorId);
+        evse.tryPlug(connectorId);
```

Comment:
I shared my opinion. I'd like simulator to behave closer to real station, then it's easier to keep concepts in mind and reproduce some real-case scenarios. User should be always able to plug cable unless it's some sophisticated station with physical mechanism to block the socket :)
If connector status is unavailable - it simply won't charge, but cable is still there (cableStatus==plugged).

Reply:
Okay, I addressed that.